### PR TITLE
enhancement: Include offline screens and speaking screens in the end to end tests

### PR DESCRIPTION
**Pull request by Rolph1**
### Overview:
This PR enhances the end to end tests to now include both the speaking screens and the offline mode screens. The logic to go from OfflineRecordingScreen to RecordingReview has also been moved to SpeakingRepository.

### Files modified:
- `EndToEndTest`: Now contains the tests for the speaking screens and the offline mode screens 
- `SpeakingRepository`, `SpeakingRepositoryRecord`, `SpeakingViewModel`, `OfflineRecordingScreen`: Have been modified because in RecordingScreen, the user needs to have recorded an audio file and have it be saved on the cache to navigate to the OfflineRecordingReviewScreen were they could rehear said recording. The logic for this was in the code for the screen and couldn't be mocked so a new variable has been introduced to SpeakingRepository to be able to mock it.

### Conclusion:
This PR brings us one step closer to finalizing the end to end tests by adding new screens to the EndToEnd file. It also enhances the testability of the offline recording screen by moving the logic to go to the next screen to the speaking repository.
TODO: Add the BOTI and statistics screens to the end to tests to finally finish it.

### CODE REVIEW

File: app/src/androidTest/java/com/github/se/orator/endtoend/EndToEndTest.kt [hunk 1]
```diff
@@ -250,5 +356,159 @@ class EndToEndAppTest {
...
+    `when`(speakingRepository.analysisState)
+        .thenReturn(MutableStateFlow(SpeakingRepository.AnalysisState.FINISHED))
+
+    speakingViewModel =
+        SpeakingViewModel(speakingRepository, apiLinkViewModel, userProfileViewModel)
+
+    composeTestRule.runOnUiThread {
+      navController?.navigate(
+          Screen.SPEAKING) // forcing back to home where the user would be once he finishes creating
+      // his account
```
Comment (MathisHage) [inline app/src/androidTest/java/com/github/se/orator/endtoend/EndToEndTest.kt:416]: Not sure this comment should be here.
```suggestion
          Screen.SPEAKING)
```

File: app/src/androidTest/java/com/github/se/orator/endtoend/EndToEndTest.kt [hunk 2]
```diff
@@ -250,5 +356,159 @@ class EndToEndAppTest {
...
+    }
+    composeTestRule.onNodeWithTag("mic_text").assertTextContains("Analysis finished.")
+
+    // Step 9: Offline mode:
+
+    composeTestRule.runOnUiThread {
+      navController?.navigate(
+          Screen.OFFLINE) // forcing back to home where the user would be once he finishes creating
+      // his account
```
Comment (MathisHage) [inline app/src/androidTest/java/com/github/se/orator/endtoend/EndToEndTest.kt:425]: Same here.
```suggestion
          Screen.OFFLINE)
```

File: app/src/androidTest/java/com/github/se/orator/endtoend/EndToEndTest.kt [hunk 3]
```diff
@@ -250,5 +356,159 @@ class EndToEndAppTest {
...
+    composeTestRule.onNodeWithTag("QuestionCard_1")
+    composeTestRule.onNodeWithTag("QuestionCard_2")
+    composeTestRule.onNodeWithTag("QuestionCard_3")
+    composeTestRule.onNodeWithTag("QuestionCard_4")
+
+    composeTestRule.onNodeWithTag("QuestionText_0")
+    composeTestRule.onNodeWithTag("QuestionText_1")
+    composeTestRule.onNodeWithTag("QuestionText_2")
+    composeTestRule.onNodeWithTag("QuestionText_3")
+    composeTestRule.onNodeWithTag("QuestionText_4")
```
Comment (MathisHage) [inline app/src/androidTest/java/com/github/se/orator/endtoend/EndToEndTest.kt:467]: Not sure for what these are used. Maybe an `assertExist()` is missing ?

**Review by MathisHage — CHANGES_REQUESTED** [review]
A few remarks regarding the tests were made. Otherwise will approve when they are addressed.

File: app/src/androidTest/java/com/github/se/orator/endtoend/EndToEndTest.kt (hunk 1 above)
Comment (Rolph1) [reply app/src/androidTest/java/com/github/se/orator/endtoend/EndToEndTest.kt:416]: oops I just copied the code to forcibly move to another screen! Great catch.

File: app/src/androidTest/java/com/github/se/orator/endtoend/EndToEndTest.kt (hunk 3 above)
Comment (Rolph1) [reply app/src/androidTest/java/com/github/se/orator/endtoend/EndToEndTest.kt:467]: Yes indeed... That's fixed now.

**Review by MathisHage — APPROVED** [review]
The changes requested were taken into account.

**Pull request merged**